controllers/component_build_controller.go

```diff
@@ -633,14 +639,20 @@ func GeneratePipelineRun(component appstudiov1alpha1.Component, bundle string, o
 		{Name: "revision", Value: tektonapi.ArrayOrString{Type: "string", StringVal: "{{revision}}"}},
 		{Name: "output-image", Value: tektonapi.ArrayOrString{Type: "string", StringVal: proposedImage}},
 	}
-	if dockerFile, err := devfile.SearchForDockerfile([]byte(component.Status.Devfile)); err == nil && dockerFile != nil {
+
+	dockerFile, err := devfile.SearchForDockerfile([]byte(component.Status.Devfile))
+	if err != nil {
+		return nil, err // devfile could be invalid to cause parser error
```

Comment:
Could you elaborate more on this?

Reply:
Good question. After rethinking, I think this comment may not be accurate to describe the possible error. Any error from `SearchForDockerfile` should cause `GeneratePipelineRun` returns immediately. This also answers your question above.